### PR TITLE
Lookup structs for Kimchi prover in o1js

### PR DESCRIPTION
**Pull request by pablodeymo**

### CODE REVIEW

**Review by xqft — APPROVED** [review]
nit: I would change docs to use the [TSDoc](https://tsdoc.org/) standard

File: verifier_circuit/src/prover/prover.ts [hunk 1]
```diff
@@ -671,14 +689,25 @@ export class RecursionChallenge {
     }
 }
 
+/// Commitments linked to the lookup feature
```
Comment (pablodeymo) [inline verifier_circuit/src/prover/prover.ts:692]:
```suggestion
/**
* Commitments linked to the lookup feature
*/
```

File: verifier_circuit/src/lookups/lookups.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+/// Flags for each of the hard-coded lookup patterns.
```
Comment (pablodeymo) [inline verifier_circuit/src/lookups/lookups.ts:2]:
```suggestion
/**
* Flags for each of the hard-coded lookup patterns.
*/
```

File: verifier_circuit/src/lookups/lookups.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export class LookupFeatures {
+    /// A single lookup constraint is a vector of lookup constraints to be applied at a row.
+    patterns: LookupPatterns
+    /// Whether joint lookups are used
+    joint_lookup_used: boolean
+    /// True if runtime lookup tables are used.
+    uses_runtime_tables: boolean
+}
+
+/// Describes the desired lookup configuration.
```
Comment (pablodeymo) [inline verifier_circuit/src/lookups/lookups.ts:19]:
```suggestion
/**
* Describes the desired lookup configuration.
*/
```

**Pull request merged**